File: app/src/main/java/com/denchic45/kts/uivalidator/validatorlistener/MessageIdLiveDataValidationListener.java
package com.denchic45.kts.uivalidator.validatorlistener;

import androidx.lifecycle.MutableLiveData;

import com.denchic45.kts.uivalidator.Rule;

public class MessageIdLiveDataValidationListener extends LiveDataValidationListener<Integer> {
    public MessageIdLiveDataValidationListener(MutableLiveData<Integer> mutableLiveData) {
        super(mutableLiveData);
    }

    @Override
    void onSuccess() {
        mutableLiveData.setValue(null);
    }

    @Override
    void onError(Rule rule) {
        mutableLiveData.setValue(rule.getErrorResId());
    }
}

File: app/src/main/java/com/denchic45/kts/uivalidator/validatorlistener/MessageIdLiveDataValidationListener.kt
package com.denchic45.kts.uivalidator.validatorlistener

import com.denchic45.kts.SingleLiveData
import com.denchic45.kts.uivalidator.Rule

class MessageIdLiveDataValidationListener(mutableLiveData: SingleLiveData<Int>) :
    LiveDataValidationListener<Int>(mutableLiveData) {
    override fun onSuccess() {
//        mutableLiveData.value = null
    }

    override fun onError(rule: Rule) {
        mutableLiveData.value = rule.errorResId
    }
}